src/Mvc/Mvc.Testing/src/DeferredHostBuilder.cs

```diff
@@ -38,8 +40,18 @@ public DeferredHostBuilder()
 
         public IHost Build()
         {
+            // Hosting configuration is being provided by args so that
+            // we can impact WebApplicationBuilder based applications.
+            var args = new List<string>();
+
+            // Transform the host configuration into command line arguments
+            foreach (var (key, value) in _hostConfiguration.AsEnumerable())
+            {
+                args.Add($"--{key}={value}");
```

Comment:
Do we need to do any fancy escaping for `value`? Like can values have spaces which would break config parsing? I'm not super familiar with this area

Reply:
I don't see how it breaks given how the [parsing works](https://github.com/dotnet/runtime/blob/cf52b7e38d28b7e28358cee052effe0ed47efb2e/src/libraries/Microsoft.Extensions.Configuration.CommandLine/src/CommandLineConfigurationProvider.cs#L44). We're not parsing the command line string, we're parsing each arg. 

I also don't see a way to escape `-` or `/` so that might be the only problem.

Reply:
An `=` in the key would also mess this up, right? I'm not sure how realistic that is.

Reply:
> I also don't see a way to escape - or / so that might be the only problem.

You could always base64 encode/decode. Stripping and adding back the `=` padding characters might be a pain though.